File: DataStructureAndAlgorithm/Searching/DFS/Python/dfs.py
def dfs(graph, start_node, visited=None):
    if visited is None:
        visited = set()
    visited.add(start_node)
    print(f"Visited {start_node}")

    for neighbor in graph.get(start_node, []):
        if neighbor not in visited:
            dfs(graph, neighbor, visited)

File: DataStructureAndAlgorithm/Searching/DFS/C++/DFS.cpp
#include "DFS.h"
#include <iostream>

void DFS(int startNode, int numNodes, const std::vector<std::list<int>>& adj, std::vector<bool>& visited) {
    visited[startNode] = true;
    std::cout << "Visited " << startNode << std::endl;

    for (int neighbor : adj[startNode]) {
        if (!visited[neighbor]) {
            DFS(neighbor, numNodes, adj, visited);
        }
    }
}
